Source/OCMockTests/OCMockObjectMacroTests.m

```diff
@@ -477,4 +477,27 @@ - (void)testCanUseMacroToStubMethodWithAnyNonObjectArgumentChainedWithOCMStubRec
     XCTAssertEqualObjects(@"f", [mock commonPrefixWithString:@"foo" options:NSCaseInsensitiveSearch]);
 }
 
+- (void)testInitWithRecorderMethodsBeingPassedThroughMock
+{
+    // Because of the way the macros work, you can call recorder methods on the mock and they
+    // will work correctly. Technically these are a mix of old syntax and new.
+    // If these test fail we expect a crash.
```

Comment:
How relevant is this? I'm asking because the implementation could be simplified significantly if we don't support this mixing of styles.

Reply:
Just looking at our codebase:
I have about 150 cases of `ignoringNonObjectArgs`.
I have about 50 cases of `classMethod`.
I have about 5 cases of andReturn/andDo/etc

We had more, but I've cleaned a lot of them up over the past couple of weeks. I'm ok with not supporting mixing the styles, but would want to see guardrails in place (preferably at compile time, but runtime is also ok) to prevent people from mixing them as opposed to just stating we don't support the mixing of the styles.

Reply:
See my comment in the main thread.